### PR TITLE
Nested Tensor + AOTI test

### PR DESCRIPTION
**Pull request by davidberard98**
Since we expect AOTI to be important for serving NJT in the future, I'm adding a test demonstrating that AOTI currently works with NJT when NJT is entirely in the graph (no NJTs going in or out), and to prevent regressions.

Fixes #ISSUE_NUMBER

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: test/inductor/test_aot_inductor.py [hunk 1]
```diff
@@ -2676,6 +2676,62 @@ def forward(self, x):
...
+
+            def forward(self, values, offsets):
+                nt = torch.nested.nested_tensor_from_jagged(values, offsets)
+                res = self.mlp(nt)
+                return res.values()
+
+        model = Model().to(device=self.device)
+
+        example_inputs_1 = (
+            torch.randn((15, 128), device=self.device),
```
Comment (jbschlosser) [inline test/inductor/test_aot_inductor.py:2695]: do you think it's valuable to also include a set of example inputs that have the same `values` shape but different ragged structure / offsets? e.g. two example inputs of shape `(15, 128)` with different `offsets` but same *number* of offsets

**Review by jbschlosser — COMMENTED** [review]
Thanks for adding this! I'm not familiar with AOTI so I'll let someone who is sign off on it, but it looks like correct usage AFAICT

File: test/inductor/test_aot_inductor.py [hunk 2]
```diff
@@ -2676,6 +2676,62 @@ def forward(self, x):
...
+
+        self.assertEqual(
+            aot_inductor_module(*example_inputs_1), model(*example_inputs_1)
+        )
+        self.assertEqual(
+            aot_inductor_module(*example_inputs_2), model(*example_inputs_2)
+        )
+        self.assertEqual(
+            aot_inductor_module(*example_inputs_3), model(*example_inputs_3)
+        )
```
Comment (angelayi) [inline test/inductor/test_aot_inductor.py:2733]: You can also use `check_model_with_multiple_inputs` ([example](https://github.com/pytorch/pytorch/blob/38278104537351c7b4683df8607057888360e759/test/inductor/test_aot_inductor.py#L667-L669))

**Review by angelayi — APPROVED** [review]
overall lgtm!

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**